CONTRIBUTING.md

```diff
@@ -1,20 +1,55 @@
-2. Gerekli değişiklikler yapıldıktan sonra ana repository'e Pull Request gönderilir. 
+1. Katkıda bulunmak isteyenler, ilgili repository'de açılan issue'lar içerisinden (https://github.com/CodeFiction/typescript-book/issues) bir görevi üstlenip, değişiklikleri ve çevirileri kendi kopyası (forku) üzerinden gerçekleştirir.
+2. Gerekli değişiklikler yapıldıktan sonra ana repository'e Pull Request gönderilir.
 	..* Ana reposory'e gelen pull request'ini **iki farklı kişi** gözden geçirir.
-	..* İki ayrı kullanıcının da onayını alan pull request sonrasında ana repository'e merge (rebase and merge yöntemiyle) edilir. 
+	..* İki ayrı kullanıcının da onayını alan pull request sonrasında ana repository'e merge (rebase and merge yöntemiyle) edilir.
 	..* Gelen pull request'i merge işlemini gerçekleştiren kişi, aynı isimle sonuna "gözden geçirilecek" ekleyerek yeni bir issue açar. *(Açılan bu issue ile amacımız, yapılan değişikliklerin Türkçe dil ve imla kurallarına uygunluğunun denetlenmesini sağlamak.)*
 
 ### Ek Notlar
-- Katkıda bulunmak istediğiniz konuya yorum bırakarak son durumu sorabilirsiniz. Katılmak için talepte bulunabilirsiniz. 
+- Katkıda bulunmak istediğiniz konuya yorum bırakarak son durumu sorabilirsiniz. Katılmak için talepte bulunabilirsiniz.
 - Atanmış bir issue ile ilgili yardımcı olabilecekseniz issue sahibi ile iletişim geçilmelisiniz.
 - Üzerinizdeki görevi(issue) bırakmanız gerekirse lütfen topluluğa ilgili Gitter kanalı üzerinden haber veriniz.
-- Teknik terimlerin çevirisinde bir bütünlük sağlamak amacı ile bir [sözlük](GLOSSARY.md) oluşturulmaktadır. Lütfen çevirinizi yaparken ve bir Pull Request'i incelerken bu sözlükte yer alan terimleri takip ediniz. 
+- Teknik terimlerin çevirisinde bir bütünlük sağlamak amacı ile bir [sözlük](GLOSSARY.md) oluşturulmaktadır. Lütfen çevirinizi yaparken ve bir Pull Request'i incelerken bu sözlükte yer alan terimleri takip ediniz.
 - Diğer görüş ve önerileriniz için gitter kanalına bekleriz. https://gitter.im/codefiction/typescript-book
 
+### Kodlarda Değişiklik Yaparken
+- Öncelikle [codefiction/typescript-book](https://github.com/CodeFiction/typescript-book) repository'sini **fork** yaparak kendinize bir kopyasını oluşturun.
+- Daha sonra `git clone` komutunu kullanarak kendi bilgisayarınıza kendinize ait olan kopyayı indirin.
+- Kopyayı indirdikten sonra `git remote` komutunu kullanarak **codefiction** repository'sini yeni bir *remote* olarak ekleyin.
+```sh
+git remote add upstream https://github.com/CodeFiction/typescript-book.git
+```
+- Değişikliklerinizi yaptıktan sonra uygun **commit mesajı** ile değiğikliklerinizi *commit* yapın.
+```sh
+git commit -m "Contributing kurallarına git mesajlarını ekliyorum"
+```
+- Yaptığınız değişiklik *push* yapmadan önce **git pull** komutunu kullanarak *upstream* i lokalinize indirin.
```

Comment:
Push işlemini ve upstream işlemini ayrıca yazabiliriz. Git deneyimi olmayan arkadaşlar için, upstream ekleme işlemi her push'tan önce yapılan bir işlem gibi algılanabilir.